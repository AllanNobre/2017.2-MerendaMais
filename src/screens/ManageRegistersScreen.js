import React from 'react';
import { StyleSheet, Text, View, ScrollView, ActivityIndicator, TouchableOpacity, Alert } from 'react-native';
import PropTypes from 'prop-types';
import Header from '../components/Header';
import { logInfo } from '../../logConfig/loggers';

const FILE_NAME = 'ManageRegistersScreen.js';

const styles = StyleSheet.create({
  listRegisters: {
    flex: 1,
    marginHorizontal: 15,
    marginVertical: 10,
    borderColor: 'black',
    borderWidth: 1,
    borderRadius: 3,
    backgroundColor: '#FAFAFA',
    justifyContent: 'space-between',
  },
  textBox: {
    paddingLeft: 2,
    justifyContent: 'flex-start',
  },
  text: {
    fontSize: 15,
    paddingVertical: 5,
  },
  buttonBox: {
    flexDirection: 'row',
    justifyContent: 'center',
    margin: 10,
  },
  greenBox: {
    backgroundColor: '#4CD964',
    padding: 8,
    borderRadius: 3,
    marginRight: 20,

  },
  redBox: {
    backgroundColor: '#FF3B30',
    padding: 8,
    borderRadius: 3,
    marginRight: 10,
  },
});

export default class ManageRegistersScreen extends React.Component {
  componentWillMount() {
    this.props.asyncGetCounselorFromGroup(this.props.CAE, this.props.cpf);
  }

  componentWillReceiveProps(newProps) {
    console.log(newProps);
  }

  disableCounselor(counselor, codGroup) {
    Alert.alert(
      'Desativar Conselheiro',
      'Você deseja realmente desassociar esse Conselheiro da Aplicação?',
      [
        { text: 'Cancelar', onPress: () => console.log('Cancelar') },
        { text: 'Desassociar', onPress: () => this.props.disableCounselor(counselor, codGroup) },
      ]);
  }


  arrayRegistersList() {
    if (this.props.listOfCounselorsInAGroup.length === 0) {
      return (
        <ActivityIndicator style={{ marginTop: 50 }} size="large" color="#FF9500" />
      );
    }
    return (
      this.props.listOfCounselorsInAGroup.map(counselor => (
        <View style={styles.listRegisters}>
          <View style={styles.textBox}>
            <Text style={styles.text}>
              <Text style={{ fontWeight: 'bold' }}>Nome: </Text>
              {counselor.name}
            </Text>
            <Text style={styles.text}>
              <Text style={{ fontWeight: 'bold' }}>CPF: </Text>
              {counselor.cpf}
            </Text>
            <Text style={styles.text}>
              <Text style={{ fontWeight: 'bold' }}>Telefone: </Text>
              {counselor.phone}
            </Text>
          </View>
          <View style={styles.buttonBox}>
            <TouchableOpacity>
              <View style={styles.greenBox}>
                <Text>VALIDAR</Text>
              </View>
            </TouchableOpacity>

            <TouchableOpacity
              onPress={() => this.disableCounselor(counselor, this.props.codGroup)}
            >
              <View style={styles.redBox}>
                <Text>EXCLUIR</Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
      ))
    );
  }

  render() {
    logInfo(FILE_NAME, 'rebder',
      `Counselor List: ${this.props.listOfCounselorsInAGroup}`);
    return (
      <View style={{ backgroundColor: 'white', flex: 1 }}>
        <View>
          <Header
            title={'Gerenciar Registro'}
            backButton
          />
        </View>
        <ScrollView>
          {this.arrayRegistersList()}
        </ScrollView>
      </View>
    );
  }
}

ManageRegistersScreen.propTypes = {
  CAE: PropTypes.string.isRequired,
  cpf: PropTypes.string.isRequired,
  codGroup: PropTypes.string.isRequired,
  listOfCounselorsInAGroup: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    cpf: PropTypes.string,
    phone: PropTypes.string,
  })).isRequired,
  asyncGetCounselorFromGroup: PropTypes.func.isRequired,
  disableCounselor: PropTypes.func.isRequired,
};
