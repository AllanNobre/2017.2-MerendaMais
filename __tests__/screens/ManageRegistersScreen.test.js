import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import configureStore from 'redux-mock-store';
import ManageRegistersScreen from '../../src/screens/ManageRegistersScreen';

Enzyme.configure({ adapter: new Adapter() });

// const mockStore = configureStore();

const props = {
  listOfCounselorsInAGroup: [{ nome: 'Joao' }],
  CAE: 'RO',
  asyncGetCounselorFromGroup: jest.fn(),
};

describe('Testing ManagerRegistersScreen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <ManageRegistersScreen {...props} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
