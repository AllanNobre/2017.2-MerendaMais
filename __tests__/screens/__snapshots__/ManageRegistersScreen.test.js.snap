// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ManagerRegistersScreen renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View>
    <Header
      backButton={true}
      subTitle=""
      title="Gerenciar Registro"
    />
  </View>
  <ScrollViewMock>
    <View
      style={
        Object {
          "backgroundColor": "#FAFAFA",
          "borderColor": "black",
          "borderRadius": 3,
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "space-between",
          "marginHorizontal": 15,
          "marginVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "flex-start",
            "paddingLeft": 2,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Nome: 
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            CPF: 
          </Text>
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Telefone: 
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 10,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "backgroundColor": "#4CD964",
                "borderRadius": 3,
                "marginRight": 20,
                "padding": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              VALIDAR
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "backgroundColor": "#FF3B30",
                "borderRadius": 3,
                "marginRight": 10,
                "padding": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              EXCLUIR
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </ScrollViewMock>
</View>
`;
